org.jacoco.core.test/pom.xml

```diff
@@ -182,6 +183,87 @@
         </plugins>
       </build>
     </profile>
+    <profile>
+      <id>kotlin-validation</id>
+      <activation>
+        <!-- This profile should be activated for all jdk versions greater than 1.8 -->
+        <property>
+          <name>jdk.version</name>
+          <value>1.8</value>
+        </property>
+      </activation>
+      <properties>
+        <kotlin.version>1.2.51</kotlin.version>
+        <kotlin.compiler.jvmTarget>1.8</kotlin.compiler.jvmTarget>
+      </properties>
+      <dependencies>
+        <dependency>
+          <groupId>org.jetbrains.kotlin</groupId>
+          <artifactId>kotlin-stdlib</artifactId>
+          <version>${kotlin.version}</version>
+        </dependency>
+      </dependencies>
+      <build>
+        <plugins>
+          <plugin>
+            <artifactId>kotlin-maven-plugin</artifactId>
+            <groupId>org.jetbrains.kotlin</groupId>
+            <version>${kotlin.version}</version>
+
+            <executions>
+              <execution>
+                <id>compile</id>
+                <goals>
+                  <goal>compile</goal>
+                </goals>
+                <configuration>
+                  <sourceDirs>
+                    <sourceDir>${project.basedir}/src-kotlin</sourceDir>
+                  </sourceDirs>
+                </configuration>
+              </execution>
+            </executions>
+          </plugin>
+          <plugin>
+            <groupId>org.codehaus.mojo</groupId>
+            <artifactId>build-helper-maven-plugin</artifactId>
+            <executions>
+              <execution>
+                <id>add-source</id>
+                <phase>generate-sources</phase>
+                <goals>
+                  <goal>add-source</goal>
+                </goals>
+                <configuration>
+                  <sources>
+                    <source>src-kotlin</source>
+                  </sources>
+                </configuration>
+              </execution>
+            </executions>
+          </plugin>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-compiler-plugin</artifactId>
+            <version>2.3.2</version>
+            <executions>
+              <!-- Replacing default-compile as it is treated specially by maven -->
```

Comment:
Seems that this comment was copied from https://kotlinlang.org/docs/reference/using-maven.html#compiling-kotlin-and-java-sources And IMO description from this page

> kotlin-maven-plugin should be run before maven-compiler-plugin

 is cleaner than comment

> Replacing default-compile as it is treated specially by maven

Also page says

> To compile mixed code applications Kotlin compiler should be invoked before Java compiler. 

and I'm wondering what "mixed" means - code written in Java can invoke code written in Kotlin? or vise versa? or both?

Reply:
I think “mixed” means you can call code both ways. Also, I will update the comment